pandas_log/pandas_execution_stats.py

```diff
@@ -85,27 +97,34 @@ def persist_execution_stats(self):
         set_df_attr(self.output_df, "execution_history", prev_exec_history)
         append_df_attr(self.output_df, "execution_history", self)
 
-    def log_stats_if_needed(self, silent, verbose):
+    def log_stats_if_needed(self, silent, verbose, copy_ok):
 
         from pandas_log.pandas_log import ALREADY_ENABLED
 
         if silent or not ALREADY_ENABLED:
             return
 
-        if (
-            verbose
-            or self.fn.__name__ not in PANDAS_ADDITIONAL_METHODS_TO_OVERIDE
-        ):
-            print(self)
+        if verbose or self.fn.__name__ not in DATAFRAME_ADDITIONAL_METHODS_TO_OVERIDE:
+            s = self.__repr__(verbose, copy_ok)
+            if s:
+                # If this method isn't patched and verbose is False, __repr__ will give an empty string, which
+                # we don't want to print
+                print(s)
 
-    def get_logs_for_specifc_method(self):
+    def get_logs_for_specifc_method(self, verbose, copy_ok):
         self.fn_kwargs["kwargs"] = self.fn_kwargs.copy()
-        log_method = getattr(patched_logs_functions, "log_default")
-        with suppress(AttributeError):
+        self.fn_kwargs["copy_ok"] = copy_ok
+        try:
             log_method = getattr(
                 patched_logs_functions,
                 f"{PATCHED_LOG_METHOD_PREFIX}{self.fn.__name__}",
             )
+        except AttributeError:
+            # Method is listed as a method to override, but no patched function exists
+            if verbose:
```

Comment:
@cmdavis4 can you explain this ?

Reply:
This makes it so that verbose also controls the behavior for the methods that are listed in DATAFRAME_METHODS_TO_OVERRIDE/DATAFRAME_ADDITIONAL_METHODS_TO_OVERRID but that don't have corresponding patched functions in patched_log_functions.py. Now the log_default message only gets printed if verbose. I wanted to have the option not to print the default message since it can clutter a notebook.

Reply:
@cmdavis4 there is the silent params for that

Reply:
Silent stops all output though. My goal here was to have some way to control whether to print the "default" message for methods that are listed as methods to override, but that don't have a specific patched log function, because in my experience the default message doesn't provide much useful information and can rapidly get printed many times, eg by pandas methods that call other pandas methods. If you have another suggestion for how to do this I'm open to it. I'll also take it out if that's blocking getting this merged.